docs/sources/flow/reference/components/otelcol.processor.resourcedetection.md

```diff
@@ -0,0 +1,940 @@
+in a format that conforms to the [OpenTelemetry resource semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/resource/semantic_conventions/), and appends or
+overrides the resource values in the telemetry data with this information.
+
+{{% admonition type="note" %}}
+`otelcol.processor.resourcedetection` is a wrapper over the upstream
+OpenTelemetry Collector Contrib `resourcedetection` processor. If necessary, 
+bug reports or feature requests will be redirected to the upstream repository.
+{{% /admonition %}}
+
+You can specify multiple `otelcol.processor.resourcedetection` components by giving them
+different labels.
+
+## Usage
+
+```river
+otelcol.processor.resourcedetection "LABEL" {
+  output {
+    logs    = [...]
+    metrics = [...]
+    traces  = [...]
+  }
+}
+```
+
+## Arguments
+
+`otelcol.processor.resourcedetection` supports the following arguments:
+
+Name        | Type           | Description                                                                            | Default   | Required
+----------- | -------------- | -------------------------------------------------------------------------------------- |---------- | --------
+`detectors` | `list(string)` | An ordered list of named detectors which should be ran to detect resource information. | `["env"]` | no
+`override`  | `bool`         | Configures whether existing resource attributes should be overridden or preserved.     | `true`    | no
+`timeout`   | `duration`     | Timeout by which all specified detectors must complete.                                | `"5s"`    | no
+
+`detectors` could contain the following values:
+* `env`
+* `ec2`
+* `ecs`
+* `eks`
+* `elasticbeanstalk`
+* `lambda`
+* `azure`
+* `aks`
+* `consul`
+* `docker`
+* `gcp`
+* `heroku`
+* `system`
+* `openshift`
+* `kubernetes_node`
+
+Note that `env` is the only detector which is not configured through a River block.
+The `env` detector reads resource information from the `OTEL_RESOURCE_ATTRIBUTES` environment variable. 
+This is expected to be in the format `<key1>=<value1>,<key2>=<value2>,...`, 
+the details of which are currently pending confirmation in the OpenTelemetry specification.
+
+If a detector other than `env` is needed, it can be customized by using the relevant River block for it.
+For example, the `ec2` detector can be customized using the [ec2][] block.
+If the [ec2][] block is omitted, the defaults specified in the [ec2][] block documentation will be used.
+
+If multiple detectors are inserting the same attribute name, the first detector to insert wins. 
+For example if you had `detectors = ["eks", "ec2"]` then `cloud.platform` will be `aws_eks` instead of `ec2`. 
+
+For AWS, the below ordering is recommended:
+  1. [lambda][]
+  1. [elasticbeanstalk][]
+  1. [eks][]
+  1. [ecs][]
+  1. [ec2][]
+<!-- TODO: Why do the otel docs also list a GCP ordering? There is only one GCP detector. -->
```

Comment:
I need to check this:
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.87.0/processor/resourcedetectionprocessor/README.md#gcp

Reply:
the resourcedetection readme needs a fix probably. I found that gce and gke were deprecated to use only one gcp detector https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/10347, ordering for gcp is not needed anymore if I understood correctly